pandas/tests/io/test_gcs.py

```diff
@@ -26,6 +26,21 @@ def test_read_csv_gcs(mock):
     assert_frame_equal(df1, df2)
 
 
+@td.skip_if_no('gcsfs')
+def test_to_csv_gcs(mock):
+    df1 = DataFrame({'int': [1, 3], 'float': [2.0, np.nan], 'str': ['t', 's'],
+                     'dt': date_range('2018-06-18', periods=2)})
+    with mock.patch('gcsfs.GCSFileSystem') as MockFileSystem:
```

Comment:
Might be missing the point but if you are patching this what is actually getting tested for gcs?

Reply:
Yeah, it's kind of the same problem that we discussed in #20729. This does at least test the logic that I touched here; I think all the mocks assume is that `gcsfs.GCSFileSystem` can `read`/`write` strings and everything else is using the real `pandas` methods.